components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -25,6 +52,56 @@ BraveWalletProviderImpl::BraveWalletProviderImpl(
 
 BraveWalletProviderImpl::~BraveWalletProviderImpl() {}
 
+void BraveWalletProviderImpl::AddEthereumChain(
+    std::vector<mojom::EthereumChainPtr> chains,
+    AddEthereumChainCallback callback) {
+  if (!delegate_ || chains.empty()) {
+    RespondForEthereumChainRequest(
+        std::move(callback),
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
+  DCHECK_LT(chains.size(), size_t(2));
+  const auto& chain = chains.front();
```

Comment:
The method accepts a **single** object parameter, with a chainId and some chain metadata. The method returns null if the chain was added to the wallet, and an error otherwise.

`wallet_addEthereumChain` accepts a **single** object parameter, specified by the following TypeScript interface:
```
interface AddEthereumChainParameter {
  chainId: string;
  blockExplorerUrls?: string[];
  chainName?: string;
  iconUrls?: string[];
  nativeCurrency?: {
    name: string;
    symbol: string;
    decimals: number;
  };
  rpcUrls?: string[];
}
```
The spec says single object parameter and the RPC params always takes array so for the perspective of `AddEthereumChain` API, we should always have a single `mojom::EthereumChainPtr`

Reply:
`params` passed as array and we should parse it as array. We take only the first and single element from parameters now, seems ok?

Reply:
No, the RPC interfaces defines array for `params` because some methods might requires multiple params.
This is not the parsing end of json RPC([BraveWalletJSHandler::Request](https://github.com/brave/brave-core/pull/9798/files#diff-0bcf06f5f27be321cfd97938041eef610ff08c29a35ffce39ba62f564322f104R358-R364) already handles the params and we only takes first one there), it is the API we really use to add a chain data and by spec it is single parameter.